tests/test_slips_utils.py

```diff
@@ -1,10 +1,410 @@
+    "input_value, expected_output",
+    [
+        (0.1, "low"),
+        (0.4, "medium"),
+        (0.5, "medium"),
+        (0.7, "high"),
+        (1.0, "critical"),
+        (-1, "info"),
+        (2, None),
+    ],
+)
+def test_threat_level_to_string(input_value, expected_output):
+    # utils = ModuleFactory().create_utils_obj()
+    ...
+
+
+def test_is_private_ip():
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.is_private_ip(ipaddress.ip_address("192.168.1.1"))
+    assert utils.is_private_ip(ipaddress.ip_address("10.0.0.1"))
+    assert utils.is_private_ip(ipaddress.ip_address("172.16.0.1"))
+    assert not utils.is_private_ip(ipaddress.ip_address("8.8.8.8"))
+    assert not utils.is_private_ip(ipaddress.ip_address("0.0.0.0"))
+    assert not utils.is_private_ip(ipaddress.ip_address("255.255.255.255"))
+
+
+@pytest.mark.parametrize(
+    "timestamp, expected_result",
+    [
+        ("1680788096.789", "1680788096"),
+        ("1680788096", "1680788096"),
+        ("1680788096.", "1680788096"),
+    ],
+)
+def test_remove_milliseconds_decimals(timestamp, expected_result):
+    utils = ModuleFactory().create_utils_obj()
+    result = utils.remove_milliseconds_decimals(timestamp)
+    assert (
+        result == expected_result
+    ), f"Expected {expected_result}, but got {result}"
+
+
+@pytest.mark.parametrize(
+    "start_time, end_time, return_type, expected_result",
+    [
+        (1609459200, 1609545600, "days", 1),
+        (1609459200, 1609545600, "hours", 24),
+        (1609545600, 1609459200, "seconds", lambda result: result < 0),
+        (1609459200.0, 1609459200.1, "seconds", 0.1),
+    ],
+)
+def test_get_time_diff(start_time, end_time, return_type, expected_result):
+    utils = ModuleFactory().create_utils_obj()
+    result = utils.get_time_diff(start_time, end_time, return_type)
+    if callable(expected_result):
+        assert expected_result(
+            result
+        ), f"Expected the result to satisfy the condition, but got {result}"
+    else:
+        assert (
+            result == expected_result
+        ), f"Expected {expected_result}, but got {result}"
+
+
+@pytest.mark.parametrize(
+    "seconds, expected_timedelta",
+    [
+        (3600, datetime.timedelta(seconds=3600)),
+        (86400, datetime.timedelta(days=1)),
+        (-3600, datetime.timedelta(seconds=-3600)),
+        (31536000, datetime.timedelta(days=365)),
+    ],
+)
+def test_to_delta(seconds, expected_timedelta):
+    utils = ModuleFactory().create_utils_obj()
+    result = utils.to_delta(seconds)
+    assert (
+        result == expected_timedelta
+    ), f"Expected {expected_timedelta}, but got {result}"
+
+
+@pytest.mark.parametrize(
+    "side_effect, expected_result",
+    [
+        (None, lambda utils: isinstance(utils.get_own_IPs(), list)),
+        (
+            requests.exceptions.ConnectionError,
+            lambda utils: "127.0.0.1" in utils.get_own_IPs()
+            or not utils.get_own_IPs(),
```

Comment:
is it possible to simplify these lambda functions? they're very hard to read. for example we can extract them to a function, and give ti a descriptive name and docstring to be able to get what they're doing?